### PR TITLE
Drinking it all up mechanic.

### PR DESCRIPTION
**Pull request by LudwigVonChesterfield**
A mechanic that would allow to drink the entire bottle/glass up, in one go. To do so, use the verb in right click.

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 1]
```diff
@@ -7,6 +7,26 @@
 	unacidable = 1 //glass
 	center_of_mass = list("x"=16, "y"=10)
 
+/obj/item/weapon/reagent_containers/food/drinks/standard_feed_mob(var/mob/user, var/mob/target)
+	// Special snowflake code inbound.
+	if(is_open_container() && user == target && user.a_intent != I_HELP && user.a_intent != I_HURT) // This is a pretty specific case.
```
Comment (NanakoAC) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: this is not really a specific case, and will be frequently done by accident. 

Make it a rightclick verb instead please

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 2]
```diff
@@ -7,6 +7,26 @@
...
+			if(!H.check_has_mouth())
+				user << "Where do you intend to put \the [src]? You don't have a mouth!"
+				return
+			var/obj/item/blocked = H.check_mouth_coverage()
+			if(blocked)
+				user << SPAN_WARNING("\The [blocked] is in the way!")
+				return
+
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN) //puts a limit on how fast people can eat/drink things
+		user.visible_message(SPAN_NOTICE("[user] gulped down the whole [src]!"),SPAN_NOTICE("You gulped down the whole [src]!"))
```
Comment (NanakoAC) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: This needs a do_after of about 5-8 seconds

and if interrupted the whole drink should spill all over the user's turf

Comment (LudwigVonChesterfield) [reply code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: Will do.

Comment (NanakoAC) [reply code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm]: Right, not a preset time, base it on the quantity of reagents

do_after(user, reagents.total_volume)

big bottles will take a lot, small quantites will take almost nothing

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 3]
```diff
@@ -78,6 +78,39 @@
...
+			var/mob/living/carbon/human/H = usr
+			if(!H.check_has_mouth())
+				H << "Where do you intend to put \the [src]? You don't have a mouth!"
+				return
+			var/obj/item/blocked = H.check_mouth_coverage()
+			if(blocked)
+				H << SPAN_WARNING("\The [blocked] is in the way!")
+				return
+
+		usr.visible_message(SPAN_NOTICE("[usr] prepares to gulp down [src]."), SPAN_NOTICE("You prepare to gulp down [src]."))
```
Comment (NanakoAC) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: Some more tweaks needed here;

Only show the prepare message if the swallowing is going to take a while. I'd say 2.5 seconds. So, check if the volume is > 25

We need a visible message describing what happens if its interrupted. Or does standard_splash_mob already trigger one? What happens there

Comment (LudwigVonChesterfield) [reply code/modules/reagents/reagent_containers/food/drinks.dm]: It does trigger one. It says something like "blahblah has been splashed by blahblah".

**Comment by LudwigVonChesterfield** [comment]
I feel like it's done.

**Pull request merged**